File: Leetcode/2367. Number of Arithmetic Triplets/solution.go
func arithmeticTriplets(nums []int, diff int) int {
    var ans = 0
    var seen = make(map[int]bool)

    for _, num := range nums {
        if seen[num - diff] && seen[num - diff - diff] {
            ans += 1
        } 
        seen[num] = true
    }

    return ans
}

File: Leetcode/2367. Number of Arithmetic Triplets/solution.py
class Solution:
    def arithmeticTriplets(self, nums: List[int], diff: int) -> int:
        nums_set = set(nums)
        ans = 0
        for num in nums_set:
            if (num - diff) in nums_set and (num - (2*diff)) in nums_set:
                ans += 1
        
        return ans